src/refmvs.rs

```diff
@@ -362,15 +363,35 @@ pub(crate) struct Rav1dRefmvsDSPContext {
 impl Rav1dRefmvsDSPContext {
     pub unsafe fn splat_mv(
         &self,
-        r: &mut [refmvs_block],
+        rf: &RefMvsFrame,
         rt: &refmvs_tile,
         rmv: &Align16<refmvs_block>,
         b4: Bxy,
         bw4: usize,
         bh4: usize,
     ) {
-        let mut r = rt.r_ptrs_mut(r);
-        let rr = &mut r[(b4.y as usize & 31) + 5..];
+        let start = (b4.y as usize & 31) + 5;
+        let bx4 = b4.x as usize;
+        let mut r: [_; 37] = array::from_fn(|i| {
+            if i < start {
+                return Default::default();
+            }
+            let ri = rt.r[i];
+            if ri > rf.r.len() {
+                return Default::default();
+            }
+            // This is the range that will actually be accessed,
+            // but `splat_mv` expects a pointer offset `bx4` backwards.
+            rf.r.index_mut(ri + bx4..ri + bx4 + bw4)
+        });
+        let mut r: [_; 37] = array::from_fn(|i| {
```

Comment:
This drops the guard by re-using the name, right? Shouldn't we hold the guard across the call to `splat_mv`?

Reply:
No, `array::from_fn` creates a new array without consuming the old one, unlike `.map`.